src/components/ui4/RowUi4.tsx

```diff
@@ -0,0 +1,155 @@
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
+import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'
+import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons'
+
+import { useHandler } from '../../hooks/useHandler'
+import { lstrings } from '../../locales/strings'
+import { triggerHaptic } from '../../util/haptic'
+import { showError, showToast } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+
+const textHeights = {
+  small: 2,
+  medium: 3,
+  large: 0
+}
+
+export type RowType = 'copy' | 'editable' | 'questionable' | 'loading' | 'default' | 'touchable' | 'delete'
+
+interface Props {
+  body?: string
+  children?: React.ReactNode
+  error?: boolean
+  onLongPress?: () => Promise<void> | void
+  onPress?: () => Promise<void> | void
+  title?: string
+  type?: RowType
+  maximumHeight?: 'small' | 'medium' | 'large'
+  icon?: React.ReactNode
+}
+
+export const RowUi4 = (props: Props) => {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const { body, title, children, maximumHeight = 'medium', error, icon, onLongPress, onPress } = props
+  const { type = onLongPress == null && onPress == null ? 'default' : 'touchable' } = props
+
+  const numberOfLines = textHeights[maximumHeight]
+
+  const handlePress = useHandler(async () => {
+    if (type === 'copy' && body != null) {
+      triggerHaptic('impactLight')
+      Clipboard.setString(body)
+      showToast(lstrings.fragment_copied)
+    } else if (onPress != null) {
+      triggerHaptic('impactLight')
+      try {
+        await onPress()
+      } catch (err) {
+        showError(err)
+      }
+    }
+  })
+
+  const handleLongPress = useHandler(async () => {
+    if (onLongPress != null) {
+      triggerHaptic('impactLight')
+      try {
+        await onLongPress()
+      } catch (err) {
+        showError(err)
+      }
+    }
+  })
+
+  return type === 'loading' ? (
+    <View style={styles.container}>
+      <View style={styles.content}>
+        <EdgeText style={styles.textHeader}>{title}</EdgeText>
+        <ActivityIndicator style={styles.loader} color={theme.primaryText} size="large" />
+      </View>
+    </View>
+  ) : (
+    <TouchableHighlight
+      accessible={false}
+      onPress={handlePress}
+      onLongPress={handleLongPress}
+      disabled={handlePress == null && handleLongPress == null}
+      underlayColor={theme.touchHighlightUi4}
+    >
+      <View style={styles.container}>
+        {icon && <View style={styles.iconContainer}>{icon}</View>}
+        <View style={styles.content}>
+          {title == null ? null : (
+            <EdgeText disableFontScaling ellipsizeMode="tail" style={error ? styles.textHeaderError : styles.textHeader}>
+              {title}
+            </EdgeText>
+          )}
+          {children == null
+            ? typeof body === 'string' && (
```

Comment:
`body == null ? null : <EdgeText .../>`. We don't need to check for string-ness, since TypeScript does that already.